components/brave_ads/browser/ads_service_impl.cc

```diff
@@ -1683,6 +1684,32 @@ void AdsServiceImpl::OnPrefsChanged(
     } else {
       Stop();
     }
+
+    // Record P3A.
+    using brave_rewards::AdsP3AState;
+    const bool rewards_enabled =
+        profile_->GetPrefs()->GetBoolean(
+            brave_rewards::prefs::kBraveRewardsEnabled);
+    if (pref == prefs::kEnabled) {
+      if (profile_->GetPrefs()->GetBoolean(prefs::kEnabled)) {
+        brave_rewards::RecordAdsState(AdsP3AState::kAdsEnabled);
+        profile_->GetPrefs()->SetBoolean(prefs::kBraveAdsWereDisabled, false);
```

Comment:
why would we ever set this to false when the default is `false`? Since it's "at least once" it doesn't seem like we should ever have to explicitly set it to `false`

Reply:
Here we move between A4 to A2. If we keep it as 'true', we will not catch e.g.  A4 -> A2 -> A5
```
Q17: Have you enabled Brave Ads? 
A1: No wallet created
A2: Wallet created, ads not enabled
A3: Wallet created and ads enabled 
A4. Wallet created, ads enabled, then disabled, while Brave Rewards remains on
A5: Wallet created, ads enabled, then disabled, Brave Rewards disabled
```